### PR TITLE
Issue25483 tob pipeline options

### PR DESCRIPTION
**Pull request by Malarg**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: learning/tour-of-beam/frontend/lib/components/pipeline_options/button.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            widget.controller?.selectedExample?.pipelineOptions;
+
+        if (pipelineOptions == null || pipelineOptions.isEmpty) {
+          return const SizedBox.shrink();
+        }
+
+        return AppDropdownButton(
+          buttonPadding: const EdgeInsets.all(BeamSpacing.small),
+          buttonText: Text(
+            'Pipeline options (${_getPipelineOptionsCount(pipelineOptions)})',
```
Comment (alexeyinkin) [inline learning/tour-of-beam/frontend/lib/components/pipeline_options/button.dart:53]: To easy_localization with a named param.

File: learning/tour-of-beam/frontend/lib/components/pipeline_options/button.dart [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            style: Theme.of(context).textTheme.titleMedium,
+          ),
+          createDropdown: (close) {
+            return PipelineOptionsDropdownBody(
+              pipelineOptions: pipelineOptions,
+              close: close,
+            );
+          },
+          width: 300,
```
Comment (alexeyinkin) [inline learning/tour-of-beam/frontend/lib/components/pipeline_options/button.dart:62]:
```suggestion
          width: _width,
```

File: learning/tour-of-beam/frontend/lib/components/scaffold.dart [hunk 3]
```diff
@@ -25,15 +25,18 @@ import '../state.dart';
 import 'footer.dart';
 import 'login/button.dart';
 import 'logo.dart';
+import 'pipeline_options/button.dart';
 import 'profile/avatar.dart';
 import 'sdk_dropdown.dart';
 
 class TobScaffold extends StatelessWidget {
   final Widget child;
+  PlaygroundController? playgroundController;
```
Comment (alexeyinkin) [inline learning/tour-of-beam/frontend/lib/components/scaffold.dart:34]:
```suggestion
  final PlaygroundController? playgroundController;
```

**Review by alexeyinkin — CHANGES_REQUESTED** [review]

File: learning/tour-of-beam/frontend/lib/components/scaffold.dart (hunk 3 above)
Comment (Malarg) [reply learning/tour-of-beam/frontend/lib/components/scaffold.dart:34]: fixed

File: learning/tour-of-beam/frontend/lib/components/pipeline_options/button.dart (hunk 1 above)
Comment (Malarg) [reply learning/tour-of-beam/frontend/lib/components/pipeline_options/button.dart]: done

File: learning/tour-of-beam/frontend/lib/components/pipeline_options/button.dart [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import 'package:playground_components/playground_components.dart';
+
+import 'dropdown_body.dart';
+
+class PipelineOptionsButton extends StatefulWidget {
+  final PlaygroundController? controller;
```
[the remark this answers is not in the document]
Comment (Malarg) [reply learning/tour-of-beam/frontend/lib/components/pipeline_options/button.dart]: done

File: learning/tour-of-beam/frontend/assets/translations/en.yaml [hunk 5]
```diff
@@ -40,6 +40,7 @@ pages:
   tour:
     completeUnit: Complete Unit
     summaryTitle: Table of Contents
+    pipelineOptionsButtonTitle: 'Pipeline Options: ({count})'
```
Comment (alexeyinkin) [inline learning/tour-of-beam/frontend/assets/translations/en.yaml:43]:
```suggestion
    pipelineOptionsButtonTitle: 'Pipeline Options ({count})'
```

**Review by alexeyinkin — COMMENTED** [review]
LGTM (Internal review).

**Pull request merged**